Core/GDCore/IDE/WholeProjectRefactorer.cpp

```diff
@@ -1599,6 +1599,54 @@ void WholeProjectRefactorer::GlobalObjectOrGroupRemoved(
   }
 }
 
+void WholeProjectRefactorer::RemoveLayer(gd::Project &project,
+                                         gd::Layout &layout,
+                                         const gd::String &layerName) {
+  if (layerName.empty())
+    return;
+
+  layout.GetInitialInstances().RemoveAllInstancesOnLayer(layerName);
+
+  std::vector<gd::String> externalLayoutsNames =
+      GetAssociatedExternalLayouts(project, layout);
+  for (gd::String name : externalLayoutsNames) {
+    auto &externalLayout = project.GetExternalLayout(name);
+    externalLayout.GetInitialInstances().RemoveAllInstancesOnLayer(layerName);
+  }
+}
+
+void WholeProjectRefactorer::MergeLayers(gd::Project &project,
```

Comment:
If i understand correctly, you suggest that we should maybe change instances Z order so that, if the origin layer was above the target layer, we add the max Z order of the target layer to all instances coming from the origin layer so that they stay above the target layer instances visually?
I think you can add a TODO to add a parameter to have this logic implemented.